tests/unit/hooks/test_snapper.py

```diff
@@ -0,0 +1,265 @@
+            }
+        )
+    )
+    flexmock(module).should_receive('execute_command_and_capture_output').once().with_args(
+        ["snapper", "--jsonout", "-c", "foo", "list", "--disable-used-space"]
+    ).and_return(
+        json.dumps(
+            {
+                "foo": [
+                    {
+                        "subvolume": "/foo",
+                        "number": 17,
+                        "default": False,
+                        "active": False,
+                        "type": "single",
+                        "pre-number": None,
+                        "date": "2023-03-02 21:31:44",
+                        "user": "root",
+                        "cleanup": "",
+                        "description": "current",
+                        "userdata": None,
+                    },
+                ]
+            }
+        )
+    )
+    flexmock(module.Path).should_receive("exists").once().and_return(True)
+
+    src_dirs_before = ["/foo"]
+    src_dirs_after = module.prepare_source_directories(
+        {"include": "all"}, log_prefix, src_dirs_before
+    )
+    assert src_dirs_after == ["/foo/.snapshots/17/snapshot"]
+
+
+@pytest.mark.parametrize(
+    "config,src_dirs_in,expected",
+    (
+        (
+            {"include": "all"},
+            ["/foo", "/bar", "/baz"],
+            [
+                "/foo/.snapshots/17/snapshot",
+                "/bar/.snapshots/1337/snapshot",
+                "/baz/.snapshots/13337/snapshot",
+            ],
+        ),
+        (
+            {"include": ["/bar", "/foo"]},
+            ["/foo", "/bar", "/baz"],
+            ["/foo/.snapshots/17/snapshot", "/bar/.snapshots/1337/snapshot", "/baz"],
+        ),
+        (
+            {"include": "all", "exclude": ["/baz"]},
+            ["/foo", "/bar", "/baz"],
+            ["/foo/.snapshots/17/snapshot", "/bar/.snapshots/1337/snapshot", "/baz"],
+        ),
+        (
+            {"include": "all", "exclude": ["/baz"]},
+            ["/foo", "/bar", "/baz"],
+            ["/foo/.snapshots/17/snapshot", "/bar/.snapshots/1337/snapshot", "/baz"],
+        ),
+        (
+            {"include": ["/foo", "/bar", "/baz"], "exclude": ["/baz"]},
+            ["/foo", "/bar", "/baz"],
+            ["/foo/.snapshots/17/snapshot", "/bar/.snapshots/1337/snapshot", "/baz"],
+        ),
+        (
+            {"include": ["/foo", "/foo", "/foo"], "exclude": ["/baz"]},
+            ["/foo", "/bar", "/baz"],
+            ["/foo/.snapshots/17/snapshot", "/bar", "/baz"],
+        ),
+        (
+            {"include": "all", "exclude": ["/foo", "/bar", "/baz"]},
+            ["/foo", "/bar", "/baz"],
+            ["/foo", "/bar", "/baz"],
+        ),
+        (
+            {"include": "all"},
+            ["/foo", "/bar", "/baz", "/quaz"],
+            [
+                "/foo/.snapshots/17/snapshot",
+                "/bar/.snapshots/1337/snapshot",
+                "/baz/.snapshots/13337/snapshot",
+                "/quaz",
```

Comment:
I always understood `quux` to be next after `baz`. Different schools of thought I guess. :smile: